reporter/iface.go

```diff
@@ -33,12 +33,17 @@ type TraceReporter interface {
 	ReportTraceEvent(trace *libpf.Trace, meta *samples.TraceEventMeta) error
 }
 
+type FileOpener interface {
+	// OpenFile opens the given file from the process namespace.
+	OpenFile(file string) (process.ProcessFile, error)
+}
```

Comment:
I think this is useless. Are you anticipating some other implementations for the interface that would/could not implement the full `process.Process` interface?